certbot/cli.py

```diff
@@ -686,7 +675,7 @@ def prepare_and_parse_args(plugins, args, detect_defaults=False):  # pylint: dis
               "which ones are required if it finds one missing")
     helpful.add(
         None, "--dialog", dest="dialog_mode", action="store_true",
-        help="Run using interactive dialog menus")
+        help="Deprecated - this has no effect")
```

Comment:
I think we should suppress help output here rather than continuing to print messages about a deprecated flag. We have a utility function for deprecating arguments called [add_deprecated_argument](https://github.com/certbot/certbot/blob/master/certbot/util.py#L369) that does this. I think the only reason we may not want to use this is it will cause a warning to be printed whenever the flag is used. This might be overkill for flags like `--text` that might be used in renewal scripts.

If we don't want to print a warning, we should use `argparse.SUPPRESS` here as it's done in `add_deprecated_argument`.